### PR TITLE
Mortgage Performance: create states from file

### PR DESCRIPTION
**Pull request by wpears**
Depends on https://github.com/cfpb/consumerfinance.gov/pull/8080

Currently, the script that processes mortgage data works by assuming there are correctly created `State` objects in the database. However, these state objects have fields that include counties and MSAs (or non-msa counties) which are subject to change.

This PR adds a `load_states` method (along with applicable data) which derives the necessary `State` objects from checked-in files. Thus, any, eg, FIPS code updates also flow to the `State` objects and the mortgage aggregates.

## Testing

- Pull
- Run the `process_mortgage_data` script (I invoked it with `./cfgov/manage.py runscript process_mortgage_data --script-args 2023-06-01`)
- Both states and counties should log as loaded. You can exit out of the script after that check or continue until it fails on AWS credential errors

### CODE REVIEW

File: cfgov/data_research/mortgage_utilities/fips_meta.py [hunk 1]
```diff
@@ -330,6 +330,50 @@ def load_fips_meta(counties=True):
     load_constants()
 
 
+def load_states():
+    """
+    Load state objects from csv data
+    """
+    State.objects.all().delete()
```
Comment (chosak) [inline cfgov/data_research/mortgage_utilities/fips_meta.py:337]: You might consider wrapping this method (or even the whole script?) in [an atomic transaction](https://docs.djangoproject.com/en/3.2/topics/db/transactions/#django.db.transaction.atomic):

```py
from django.db import transaction

with transaction.atomic():
    # do stuff
```

That way, if something fails partway through, you're not left in a broken state.

Comment (wpears) [reply cfgov/data_research/mortgage_utilities/fips_meta.py:339]: Added in a1a015c

File: cfgov/data_research/mortgage_utilities/fips_meta.py [hunk 2]
```diff
@@ -330,6 +331,51 @@ def load_fips_meta(counties=True):
...
+        reader = csv.DictReader(f)
+        for row in list(reader):
+            states[row["abbr"]] = {
+                "fips": row["fips"],
+                "abbr": row["abbr"],
+                "name": row["name"],
+                "counties": [],
+                "msas": [],
+                "non_msa_counties": [],
+            }
```
Comment (higs4281) [inline cfgov/data_research/mortgage_utilities/fips_meta.py:352]:
```suggestion
        for row in list(reader):
            if row["abbr"] not in NON_STATES:
                states[row["abbr"]] = {
                    "fips": row["fips"],
                    "abbr": row["abbr"],
                    "name": row["name"],
                    "counties": [],
                    "msas": [],
                    "non_msa_counties": [],
                }
```

Suggestion to keep Puerto Rico out of the states list, unless policies have changed.

**Review by higs4281 — COMMENTED** [review]
It appears Puerto Rico got into the states CSV, which causes an error when running load_national_values, because the Puerto Rico StateMortgageData object will have null count fields. 

I added a suggested fix for load_states()

**Comment by wpears** [comment]
>It appears Puerto Rico got into the states CSV, which causes an error when running load_national_values, because the Puerto Rico StateMortgageData object will have null count fields.

Strange, I didn't encounter this. I checked the code and thought that the state accesses were all protected by excluding NON_STATES but I must have missed something.

**Comment by higs4281** [comment]
The error I encountered:
```bash
(cfgov) ➜  cfgov git:(mpt/create-states-from-file) ./cfgov/manage.py runscript process_mortgage_data --script-args 2023-06-01
/Users/higginsw/.pyenv/versions/cfgov/lib/python3.8/site-packages/wagtail/contrib/modeladmin/apps.py:15: RemovedInWagtail60Warning: wagtail.contrib.modeladmin is deprecated. Use wagtail.snippets or the separate wagtail-modeladmin package instead.
  warn(
States loaded
Counties loaded
..........
100000
..........
200000
..........
300000
..........
400000
..........
500000
........
process_mortgage_data took 0:11:04.691995 to create 582934 countymortgage records
Sampling dates updated; the 186 dates now range from 2008-01-01 to 2023-06-01

Dade and Miami-Dade values merged.
472 counties of 3148 were found to be valid -- 15.0%)
Aggregating data for 2008-01-01
Exception while running run() in 'data_research.scripts.process_mortgage_data'
Traceback (most recent call last):
  File "./cfgov/manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/higginsw/.pyenv/versions/cfgov/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
...
  File "/Users/higginsw/Projects/cfgov/cfgov/data_research/scripts/process_mortgage_data.py", line 169, in run
    load_mortgage_aggregates.run()
  File "/Users/higginsw/Projects/cfgov/cfgov/data_research/scripts/load_mortgage_aggregates.py", line 128, in run
    load_national_values(date)
  File "/Users/higginsw/Projects/cfgov/cfgov/data_research/scripts/load_mortgage_aggregates.py", line 99, in load_national_values
    record.aggregate_data()
  File "/Users/higginsw/Projects/cfgov/cfgov/data_research/models.py", line 337, in aggregate_data
    count_fields[field] += getattr(state, field)
TypeError: unsupported operand type(s) for +=: 'int' and 'NoneType'
```

**Comment by wpears** [comment]
>The error I encountered (I had to set fake AWS keys to get this much to run)

Aha! Yep, that makes sense why that would throw. Thanks for finding that!

**Review by higs4281 — APPROVED** [review]
I can now make it through all aggregations to a failing point on `Exporting public CSVs to S3` – which is expected.

Thanks for this improvement!

**Pull request merged**